src/components/Forms/BaseInput.jsx

```diff
@@ -22,9 +22,10 @@ export const BaseInput = ({ as: Tag, size = "medium", isError, className, isRequ
                         : "border-gray-light focus:border-primary focus:ring-0 focus:ring-primary",
                     className,
                 )}
+                value={value}
                 {...rest}
             />
-            {isRequired && <Dot className="absolute right-3" color="danger" />}
+            {isRequired && !value && <Dot className="absolute right-3" color="danger" />}
```

Comment:
Add a `propType` for it below. Also make it `isUndefined(value)` to ensure there are no other typecasting issues incase user sends `value={0}`

Reply:
Added propType and isEmpty from lodash witch check for undefined and empty string